core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -435,6 +435,24 @@ default ByteBufAllocator alloc() {
     @MustBeClosed
     SafeCloseable push();
 
+    /**
+     * Immediately run a given {@link Runnable} with this context.
+     */
+    default void run(Runnable runnable) {
+        try (SafeCloseable ignored = push()) {
```

Comment:
Could we check if `runnable` is `null` using `requireNonNull(...)` before pushing a context?